resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/SparkRackResolver.scala

```diff
@@ -72,8 +72,6 @@ private[spark] class SparkRackResolver(conf: Configuration) extends Logging {
     val rNameList = dnsToSwitchMapping.resolve(hostNames.toList.asJava).asScala
     if (rNameList == null || rNameList.isEmpty) {
       hostNames.foreach(nodes += new NodeBase(_, NetworkTopology.DEFAULT_RACK))
-      logInfo(s"Got an error when resolving hostNames. " +
```

Comment:
Good log is also essential. Could you lower this to `logDebug` instead of removing?

Reply:
yeah I don't think we want to delete this log line.  In fact I'd keep it at INFO, but I'd prevent logging this when `hostNames.isEmpty`.  (In fact, the entire function can be a no-op when `hostNames.isEmpty`).  Its a useful log msg when there really is an error -- but when the argument is empty, its logging even with no error. sorry if my description was unclear.

(also sorry for the late comment -- I thought I posted something along these lines this morning but guess I forgot to submit it ..)

Reply:
Unless we also want to log when argument is empty list, I'd just take early return for empty list argument. Someone might think having multiple return paths is bad, but IMHO having early return is way much cleaner than going through all logic for effectively no-op.

Reply:
Hmm, misunderstood the intention. I've made the suggested change.
@HeartSaVioR this case I don't think early return needed (not considering this case, yeah I'm one of the guys who's not fan of multiple return paths).

Reply:
Wanted to highlight the intention with `if (rNameList.nonEmpty)`. It can be dropped if you think since zip with empty list remains an empty collection.

Reply:
I'd say early return really fits in this case, as @squito said `In fact, the entire function can be a no-op when hostNames.isEmpty`, but no strong opinion.

Looks like you don't need to indent one more, as it can be covered with `else if` instead of `else then if`.